imswitch/imcontrol/model/managers/DetectorsManager.py

```diff
@@ -31,27 +31,37 @@ def __init__(self, detectorInfos, updatePeriod, **lowLevelManagers):
             if not self._subManagers[detectorName].forAcquisition:
                 continue
             # Connect signals
-            self._subManagers[detectorName].sigImageUpdated.connect(
-                lambda image, init, scale, detectorName=detectorName: self.sigImageUpdated.emit(
-                    detectorName, image, init, scale, detectorName==self._currentDetectorName
+            import imswitch # FXIME: THIS IS NOT CORRECT FIX IT!!!!!!!!
+            if not IS_HEADLESS:
+                self._subManagers[detectorName].sigImageUpdated.connect(
+                    lambda image, init, scale, detectorName=detectorName: self.sigImageUpdated.emit(
+                        detectorName, image, init, scale, detectorName==self._currentDetectorName
+                    )
                 )
-            )
             self._subManagers[detectorName].sigNewFrame.connect(lambda: self.sigNewFrame.emit())
 
             # Set as default if first detector
             if self._currentDetectorName is None:
                 self._currentDetectorName = detectorName
 
         # A timer will collect the new frame and update it through the communication channel
-        self._lvWorker = LVWorker(self, updatePeriod)
-        self._thread = Thread()
-        self._lvWorker.moveToThread(self._thread)
-        self._thread.started.connect(self._lvWorker.run)
-        self._thread.finished.connect(self._lvWorker.stop)
+        if IS_HEADLESS:
+            self._lvWorker = LVWorkerNOQT(self, updatePeriod)
+            self._thread = threading.Thread(target=self._lvWorker.run)
+        else:
+            # A timer will collect the new frame and update it through the communication channel
+            self._lvWorker = LVWorker(self, updatePeriod)
+            self._thread = Thread()
+            self._lvWorker.moveToThread(self._thread)
+            self._thread.started.connect(self._lvWorker.run)
+            self._thread.finished.connect(self._lvWorker.stop)
```

Comment:
can you help me please? :)

Reply:
The if-branch where `HEADLESS = True` should be removed because we want to make sure that the previous code is compatible with the new threading framework. This means deleting the `LVWorkerNOQT` class.

Reply:
I think for this we would need to sit down together and sort out these paragraphs. I'm not 100% sure if it's all sound what I did there. All other comments raised by you address this DIY/hacked-togehter NoQT attempt. Should we meet next week or so?